File: skysail.server.queryfilter/src/io/skysail/server/queryfilter/nodes/AbstractExprNode.java
package io.skysail.server.queryfilter.nodes;

import io.skysail.server.filter.ExprNode;
import io.skysail.server.filter.FilterVisitor;
import io.skysail.server.filter.Operation;
import lombok.Getter;
import lombok.NonNull;

/**
 * Each Implementor of the class AbstactExprNode has an associated
 * Operation, passed in the constructor.
 *
 */
public abstract class AbstractExprNode implements ExprNode {

    @Getter
    private final Operation operation;

    public AbstractExprNode(@NonNull Operation op) {
        this.operation = op;
    }

    @Override
    public final Object accept(FilterVisitor visitor) {
        return visitor.visit(this);
    }
}

File: skysail.server.http.query/src/io/skysail/server/http/query/AbstractExprNode.scala
package io.skysail.server.http.query

import io.skysail.server.filter.ExprNode
import io.skysail.server.filter.FilterVisitor
import io.skysail.server.filter.Operation

abstract class AbstractExprNode(op: Operation) extends ExprNode {
  
  def accept(visitor: FilterVisitor): Object = {
    visitor.visit(this)
  }
  
  override def toString() = render()
}